app/src/main/java/com/alphelios/superrich/MainActivity.kt

```diff
@@ -20,92 +23,96 @@ class MainActivity : AppCompatActivity(R.layout.activity_main) {
             }
 
-            override fun onInAppProductsFetched(skuDetailsList: List<DataWrappers.SkuInfo>) {
-                fetchedSkuDetailsList.addAll(skuDetailsList)
-                Log.d(tag, "Retrieved SKU details list : $skuDetailsList")
+            override fun onPurchasedProductsFetched(purchases: List<PurchaseInfo>) {
+                Log.d(tag, "Retrieved all purchased products : $purchases")
             }
 
-            override fun onPurchaseAcknowledged(purchase: DataWrappers.PurchaseInfo) {
-                Log.d(tag, "onPurchaseAcknowledged")
-            }
-
-            override fun onProductsPurchased(purchases: List<DataWrappers.PurchaseInfo>) {
+            override fun onProductsPurchased(purchases: List<PurchaseInfo>) {
                 purchases.forEach {
-                    when (it.sku) {
-                        "base" -> {
-
-                        }
-                        "moderate" -> {
+                    when (it.skuId) {
+                        "no_ads" -> {
 
                         }
-                        "quite" -> {
+                        "super_sword" -> {
 
                         }
-                        "plenty" -> {
+                        "100_coins" -> {
 
                         }
-                        "subscribe" -> {
+                        "200_coins" -> {
 
                         }
-                        "yearly" -> {
 
-                        }
                     }
                 }
             }
 
-            override fun onError(
-                inAppConnector: IapConnector,
-                result: DataWrappers.BillingResponse?
-            ) {
-                Log.d(tag, "Error : ${result?.message}")
+            override fun onPurchaseAcknowledged(purchase: PurchaseInfo) {
+                Log.d(tag, "onPurchaseAcknowledged : " + purchase.skuId)
+            }
+
+            override fun onPurchaseConsumed(purchase: PurchaseInfo) {
+                Log.d(tag, "onPurchaseConsumed : " + purchase.skuId)
+            }
+
+            override fun onError(inAppConnector: IapConnector, result: BillingResponse) {
+                Log.d(tag, "Error : $result")
+
+                when (result.errorType) {
+                    CLIENT_NOT_READY -> TODO()
+                    CLIENT_DISCONNECTED -> TODO()
+                    ITEM_ALREADY_OWNED -> TODO()
+                    CONSUME_ERROR -> TODO()
+                    ACKNOWLEDGE_ERROR -> TODO()
+                    FETCH_PURCHASED_PRODUCTS_ERROR -> TODO()
+                    BILLING_ERROR -> TODO()
+                }
             }
         })
 
         binding.btPurchaseCons.setOnClickListener {
-            if (fetchedSkuDetailsList.find { it.sku == "base" } != null) {
-                iapConnector.makePurchase(this,"base")
+            if (fetchedSkuDetailsList.find { it.skuId == "base" } != null) {
+                iapConnector.purchase(this, "base")
```

Comment:
- The skuId's here need to be renamed to match the new example that you gave. 
- The README needs to be updated as well in turn.